docs/graph_engine/data_spec.rst

```diff
@@ -4,22 +4,100 @@ Preparing Graph Data
 
 DeepGNN supports both homogeneous and heterogeneous graphs. Nodes and Edges support the following attributes,
 
-  * Node/Edge Type: `int`.
+  * Node/Edge Type: `int, >= 0`.
```

Comment:
Why do we still use it in [local.py](https://github.com/microsoft/DeepGNN/blob/cdieckhaus/externalize_linear/src/python/deepgnn/graph_engine/snark/local.py#L197)?

Reply:
No actual node will have type = -1, if some node in the graph has type "-1" it should have no features it is only for edge sorting.